modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/display/context/factory/SearchBarPortletDisplayContextFactory.java

```diff
@@ -303,25 +303,28 @@ protected SearchScopePreference getSearchScopePreference(
 		Portlet headerSearchBarPortlet =
 			searchBarPrecedenceHelper.findHeaderSearchBarPortlet(themeDisplay);
 
-		if (headerSearchBarPortlet != null) {
-			Optional<PortletPreferences> headerPortletPreferencesOptional =
-				portletPreferencesLookup.fetchPreferences(
-					headerSearchBarPortlet, themeDisplay);
+		if (headerSearchBarPortlet == null) {
```

Comment:
@julschong Please also apply this pattern to `_getScopeParameterName` and `_getKeywordsParameterName`.